_posts/2020-03-18-#18113.md

```diff
@@ -0,0 +1,69 @@
+---
+layout: pr
+date: 2020-03-18
+title: "Coins: Don't allow a coin to be spent and FRESH. Improve commenting."
+pr: 18113
+authors: [jnewbery]
+components: ["utxo db and indexes"]
+host: jnewbery
+status: upcoming
+commit:
+---
+
+## Notes
+
+- The coins database is the structure in Bitcoin Core which keeps track of the
+  UTXO set. We looked at the coins database in a [previous PR Review Club
+  (#17487)](./17487). The notes for that Review Club meeting include an
+  excellent description of the caching for the coins database.
+
+- The coins database is a key-value store. It was introduced in v0.8 ([PR
+  1677](https://github.com/bitcoin/bitcoin/pull/1677)) where it was keyed by
+  transaction. Pieter Wuille describes this change in a [recent Chaincode Podcast
+  episode](https://podcast.chaincode.com/2020/01/27/pieter-wuille-1.html).
+
+- In v0.15 the coins database was changed to be keyed by transaction output
+  ([PR 10195](https://github.com/bitcoin/bitcoin/pull/10195)). For more
+  information on that change see [this blog
+  post](https://johnnewbery.com/post/whats-new-in-bitcoin-core-v0.15-pt1/).
+
+- The 'prune' in ultraprune came from the fact that once all of the outputs
```

Comment:
Afaict "ultraprune" isn't introduced in the study notes (aside from in the podcast) before being referred to here.

Reply:
Good spot. I've updated the text above.